### PR TITLE
vsup_web edited to use getHashedData

### PR DESCRIPTION
**Pull request by balcirakpeter**
- For this purpose was used method getHashedDataWithGroups.

### CODE REVIEW

File: gen/vsup_web [hunk 1]
```diff
@@ -74,59 +74,84 @@ my $groups;  # $groups->{ID}->{NAME/USERS} = name/{uco} = 1;
...
 
-	my %resourceAttributes = attributesToHash $rData->getAttributes;
-	my $relationType = $resourceAttributes{$A_R_RELATION_TYPE};
+	my $relationType = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_R_RELATION_TYPE);
 
-	my @groupsData = ($rData->getChildElements)[0]->getChildElements;
+	foreach my $groupId ($data->getGroupIdsForResource(resource => $resourceId)) {
 
-	foreach my $group (@groupsData) {
+		my $groupNasmespaceId = $data->getGroupAttributeValue(group => $groupId, attrName => $A_G_NS_ID);
```
Comment (zlamalp) [inline gen/vsup_web:83]: Typo ``Nasmespace`` -> ``Namespace``

File: gen/vsup_web [hunk 2]
```diff
@@ -74,59 +74,84 @@ my $groups;  # $groups->{ID}->{NAME/USERS} = name/{uco} = 1;
...
+		if (defined $groupNasmespaceId and defined $groupNasmespaceName) {
+			$groups->{$groupNasmespaceId}->{"NAME"} = $groupNasmespaceName;
 		}
 
-		my @membersData = ($group->getChildElements)[1]->getChildElements;
-		foreach my $member (@membersData) {
+		foreach my $memberId ($data->getMemberIdsForResourceAndGroup( resource => $resourceId, group => $groupId )) {
 
-			my %uAttributes = attributesToHash $member->getAttributes;
+			my $excluded = $data->getMemberAttributeValue( member => $memberId, attrName => $A_EXCLUDED );
```
Comment (zlamalp) [inline gen/vsup_web:95]: This should be ``getUserAttributeValue()``.

File: gen/vsup_web [hunk 3]
```diff
@@ -74,59 +74,84 @@ my $groups;  # $groups->{ID}->{NAME/USERS} = name/{uco} = 1;
...
+			my $foto = $data->getUserAttributeValue(member => $memberId, attrName => $A_FOTO);
+			my $phone = $data->getUserAttributeValue(member => $memberId, attrName => $A_PHONE);
+			my $mobile = $data->getUserAttributeValue(member => $memberId, attrName => $A_MOBILE);
+			my $emailPriv = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_PRIV);
+			my $emailPrivKOS = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_PRIV_KOS);
+			my $phonePriv = $data->getUserAttributeValue(member => $memberId, attrName => $A_PHONE_PRIV);
+			my $job = $data->getUserAttributeValue(member => $memberId, attrName => $A_JOB);
+			my $jobId = $data->getUserAttributeValue(member => $memberId, attrName => $A_JOB_ID);
+			my $ns = $data->getUserAttributeValue(member => $memberId, attrName => $A_NS);
+			my $userId = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_ID);
```
Comment (zlamalp) [inline gen/vsup_web:126]: We can use ``getUserIdForMember()`` and remove ``user:core:id`` attribute from the required.

**Review by zlamalp — CHANGES_REQUESTED** [review]
Please update also script version. Thank you.

File: gen/vsup_web (hunk 1 above)
Comment (balcirakpeter) [reply gen/vsup_web]: Fixed

File: gen/vsup_web (hunk 2 above)
Comment (balcirakpeter) [reply gen/vsup_web]: Fixed

File: gen/vsup_web (hunk 3 above)
Comment (balcirakpeter) [reply gen/vsup_web]: Fixed

**Review by zlamalp — APPROVED** [review]

**Pull request merged**